### ISSUE TITLE
Postgres TABLESAMPLE query causes internal error

### ISSUE THREAD

**Issue by efung** [sqlfluff/sqlfluff#4343]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

While running lint on a sample query using `TABLESAMPLE`, I got a runtime error.

### Expected Behaviour

sqlfluff is able to generate the lint report.

### Observed Behaviour

Runtime error when linting, with a message that this should be reported

### How to reproduce

Input SQL file (taken from the PostgreSQL [wiki](https://wiki.postgresql.org/wiki/TABLESAMPLE_Implementation)):

```sql
Select *
from foo TABLESAMPLE SYSTEM (10);    --Returns about 10% of rows in foo using SYSTEM method
```

Output:

```
WARNING    Unable to lint blah2.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/linter/runner.py", line 108, in run
    yield partial()
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/linter/linter.py", line 772, in lint_rendered
    parsed = cls.parse_rendered(rendered)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/linter/linter.py", line 390, in parse_rendered
    parsed, pvs = cls._parse_tokens(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/linter/linter.py", line 223, in _parse_tokens
    parsed: Optional[BaseSegment] = parser.parse(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/parser.py", line 39, in parse
    parsed = root_segment.parse(parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/base.py", line 1275, in parse
    self.segments = self.expand(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/base.py", line 591, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/base.py", line 1275, in parse
    self.segments = self.expand(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/base.py", line 591, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/base.py", line 1275, in parse
    self.segments = self.expand(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/base.py", line 591, in expand
    res = stmt.parse(parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/base.py", line 1211, in parse
    m = parse_grammar.match(segments=segments, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 134, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/delimited.py", line 154, in match
    match, _ = self._longest_trimmed_match(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/base.py", line 225, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/base.py", line 910, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/base.py", line 790, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 134, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 211, in match
    match, matched_option = self._match_once(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 148, in _match_once
    match, matched_option = self._longest_trimmed_match(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/base.py", line 225, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/base.py", line 910, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/base.py", line 790, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 134, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/base.py", line 886, in match
    if self.exclude.match(segments, parse_context=ctx):
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 211, in match
    match, matched_option = self._match_once(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 148, in _match_once
    match, matched_option = self._longest_trimmed_match(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/base.py", line 225, in _longest_trimmed_match
    res_match: MatchResult = matcher.match(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/base.py", line 910, in match
    resp = elem.match(segments=segments, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/base.py", line 790, in match
    m = cls.match_grammar.match(segments=segments, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/sequence.py", line 134, in match
    elem_match = elem.match(mid_seg, parse_context=ctx)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/match_wrapper.py", line 49, in wrapped_match_method
    m = func(self_cls, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/segments/ephemeral.py", line 103, in wrapped_match_method
    return func(self, segments, parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 180, in match
    available_options, _ = self._prune_options(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/anyof.py", line 78, in _prune_options
    simple = opt.simple(parse_context=parse_context)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/base.py", line 71, in wrapped_method
    result = func(self, parse_context=parse_context, **kwargs)
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/base.py", line 829, in simple
    return self._get_elem(dialect=parse_context.dialect).simple(
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/parser/grammar/base.py", line 857, in _get_elem
    return dialect.ref(self._get_ref())
  File "/opt/homebrew/lib/python3.10/site-packages/sqlfluff/core/dialects/base.py", line 278, in ref
    raise RuntimeError(
RuntimeError: Grammar refers to 'BernoulliKeywordSegment' which was not found in the postgres dialect. Perhaps specify the keyword? https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes#keywords
```

### Dialect

Postgres

### Version

sqlfluff, version 1.4.5
Python 3.10.9

### Configuration

Not using any additional configuration files

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by AidanHarveyNelson** [comment]
Hi I have taken a look at this and I have created a branch to make sure this is parsable for the select statement in postgres.

I see from the wiki you have linked it mentions this also works for Delete and Update. I have tried to run these commands listed on the wiki for these two functions and have been unable to get them to work.

Do you have sample commands that work for both update and delete?

-- Working --
![image](https://user-images.githubusercontent.com/79121034/216752116-6c83cf06-2f4a-468e-b448-b42c5ac3818d.png)

-- Syntax Error --
![image](https://user-images.githubusercontent.com/79121034/216752074-225659dd-21c9-4972-b525-124f0fb4d9bb.png)

![image](https://user-images.githubusercontent.com/79121034/216752086-bc6ea3d6-a9fd-4b42-978d-c82e1d27dbd2.png)

**Comment by efung** [comment]
> Do you have sample commands that work for both update and delete?

Sorry, I don't actually have access to a Postgres server.  I was just comparing the parsing output with various dialects.

**Comment by AidanHarveyNelson** [comment]
No problem I have put a PR up to fix the parse error on the select statement
